src/Compilers/CSharp/Portable/Binder/BinderFactory.cs

```diff
@@ -105,7 +105,7 @@ internal Binder GetBinder(SyntaxNode node, CSharpSyntaxNode memberDeclarationOpt
             // Unless this is interactive retrieving a binder for global statements
             // at the very top-level (i.e. in a completely empty file) use
             // node.Parent to maintain existing behavior.
-            if (!InScript || node.Kind() != SyntaxKind.CompilationUnit)
+            if ((!InScript || node.Kind() != SyntaxKind.CompilationUnit) && node.Parent != null)
```

Comment:
>node.Parent != null [](start = 76, length = 19)

When is this additional check needed?

Reply:
There are over a 100 tests that fail inside `NullableWalker` without that extra check. You can see them here on one of the earlier CI runs for this PR https://dev.azure.com/dnceng/public/_build/results?buildId=87553&view=ms.vss-test-web.build-test-results-tab

They were mentioned here too https://github.com/dotnet/roslyn/pull/33093#discussion_r253562102